contrib/loss_functions/SortinoLossBalance.py

```diff
@@ -0,0 +1,133 @@
+"""
+SortinoHyperOptLoss
+This module defines the alternative HyperOptLoss class which can be used for
+Hyperoptimization.
+"""
+import logging
+import os
+from datetime import datetime
+
+import numpy as np
+from pandas import DataFrame, DatetimeIndex, Timedelta, date_range
+from scipy.ndimage.interpolation import shift
+
+from freqtrade.optimize.hyperopt import IHyperOptLoss
+
+
+logger = logging.getLogger(__name__)
+
+interval = os.getenv("FQT_TIMEFRAME") or "5m"
+slippage = 0.0005
+target = 0
+annualize = np.sqrt(365 * (Timedelta("1D") / Timedelta(interval)))
+
+logger.info(f"SortinoLossBalance target is set to: {target}")
+
+
+class SortinoLossBalance(IHyperOptLoss):
+    """
+    Defines the loss function for hyperopt.
+    This implementation uses the Sortino Ratio calculation.
+    """
+
+    @staticmethod
+    def hyperopt_loss_function(
+        results: DataFrame,
+        trade_count: int,
+        min_date: datetime,
+        max_date: datetime,
+        *args,
+        **kwargs,
+    ) -> float:
+        """
+        Objective function, returns smaller number for more optimal results.
+        Uses Sortino Ratio calculation.
+        """
+        hloc = kwargs["processed"]
```

Comment:
How can you access the "processed" data? In my loss function, they are not in the kwargs.

Reply:
i don't think this is a working example as is ... 

following the PR history, it's an example to show how the data provided via the function below can / could be used.